Source/WebKit/UIProcess/mac/WKPrintingView.mm

```diff
@@ -677,6 +677,10 @@ - (void)drawPageBorderWithSize:(NSSize)borderSize
     NSRectClip(footerRect);
     _webFrame->page()->drawFooterForPrinting(*_webFrame, footerRect);
     [currentContext restoreGraphicsState];
+    
+    [currentContext saveGraphicsState];
+    _webFrame->page()->drawPageBorderForPrinting(*_webFrame, (WebCore::FloatSize)borderSize);
```

Comment:
I wonder if you need to clip to the page rect, or if something else did that for you (or maybe the page rect == the context bounds?)

Reply:
also, static_cast<WebCore::FloatSize>(...)